compiler/rustc_hir_typeck/src/coercion.rs

```diff
@@ -1604,6 +1604,7 @@ impl<'tcx, 'exprs, E: AsCoercionSite> CoerceMany<'tcx, 'exprs, E> {
                         None,
                         Some(coercion_error),
                     );
+                    fcx.check_for_range_as_method_call(&mut err, expr, found, expected);
```

Comment:
hm, any reason why this isn't living inside `emit_coerce_suggestions` with the rest of the suggestions?

Reply:
Because it otherwise also triggers for the case of `Struct { field ..Default::default() }` when the problem is a missing comma.

Reply:
hm -- I'd rather suppress that specifically (checking that the expr isn't a field assignment), because I'm suspicious of other places where we're failing to make this suggestion because they don't go through `CoerceMany`

Reply:
So have an explicit check of the `hir().parent(expr.hir_id)` to see if the current range happens to be the last field in a struct literal? It can be done.

Reply:
either last field, or a field at all, depending on how particular you want to implement it -- i don't have a strong opinion either way, if you're able to do it.